lib/issuer.js

```diff
@@ -52,9 +64,107 @@ exports.issue = async ({instance, credential}) => {
   credential.issuanceDate = date.substr(0, date.length - 5) + 'Z';
   const verifiableCredential = await vc.issue({
     credential,
-    documentLoader: brJsonldDocumentLoader.documentLoader,
+    documentLoader,
     suite
   });
 
   return verifiableCredential;
 };
+
+exports.issueNew = async ({credential, profileAgentRecord}) => {
+  const {profileAgent} = profileAgentRecord;
+  const {zcaps: {
+    userDocument: userDocZcap,
+    userKak: userKakZcap,
+  }} = profileAgent;
+
+  const invocationSigner = await profileAgents.getSigner({profileAgentRecord});
+  const {httpsAgent} = brHttpsAgent;
+  const edvDocument = new EdvDocument({
+    capability: userDocZcap,
+    client: new EdvClient({httpsAgent}),
+    keyAgreementKey: kms.createKak({invocationSigner, zcap: userKakZcap}),
+    invocationSigner
+  });
+
+  const userDoc = await edvDocument.read();
+  const {content: {zcaps: {
+    ['key-assertionMethod']: assertionMethodZcap,
+    ['credential-edv-documents']: credentialsEdvZcap,
+    ['credential-edv-hmac']: credentialsEdvHmacZcap,
+    ['credential-edv-kak']: credentialsEdvKakZcap,
+  }}} = userDoc;
+
+  if(!(assertionMethodZcap && credentialsEdvZcap && credentialsEdvHmacZcap &&
+    credentialsEdvKakZcap)) {
+    throw new BedrockError('Permission denied.', 'NotAllowedError', {
+      httpStatusCode: 400,
+      public: true,
+    });
+  }
+  const {invocationTarget: {verificationMethod}} = assertionMethodZcap;
+  const issuerKey = kms.createAsymmetricKey(
+    {invocationSigner, zcap: assertionMethodZcap});
+
+  const suite = new Ed25519Signature2018({
+    verificationMethod,
+    signer: issuerKey
+  });
+
+  // FIXME: How do is this supposed to work? Is it just a matter of
+  // removing the hash fragment?
+  const issuer = verificationMethod;
+  credential.issuer = issuer;
+
+  const verifiableCredential = await vc.issue({
+    credential,
+    documentLoader,
+    suite
+  });
+
+  const edvClient = new EdvClient({
+    httpsAgent,
+    // FIXME: REQUIRED?
+    // id: edvId,
```

Comment:
And `edvId` is not required, it just means the EdvClient will require capabilities to be passed when using it.